packages/compiler/src/server/compile-service.ts

```diff
@@ -240,33 +240,38 @@ export function createCompileService({
       }
 
       let uri = document.uri;
-      let range = Range.create(0, 0, 0, 0);
+      let range: Range | undefined;
       if (err.name === "ExternalError" && err.info.kind === "emitter" && configFilePath) {
         const emitterName = err.info.metadata.name;
-        const [yamlScript] = parseYaml(await serverHost.compilerHost.readFile(configFilePath));
-        const target = getLocationInYamlScript(yamlScript, ["emit", emitterName], "key");
-        if (target.pos === 0) {
-          log({
-            level: "debug",
-            message: `Unexpected situation, can't find emitter '${emitterName}' in config file '${configFilePath}'`,
-          });
+        if (config.file) {
+          const [yamlScript] = parseYaml(config.file.file.text);
+          range = getDiagnosticRangeInTspConfig(yamlScript, emitterName);
+        }
+
+        if (range === undefined) {
+          const clientConfigEmit = clientConfigsProvider?.config?.lsp?.emit;
+          if (clientConfigEmit && clientConfigEmit.includes(emitterName)) {
+            log({
+              level: "debug",
+              message: `The emitter '${emitterName}' causing the external error is enabled in IDE settings.`,
+            });
+          } else {
+            log({
+              level: "debug",
+              message: `The emitter '${emitterName}' causing the external error has no associated location.`,
```

Comment:
this is unexpected, you can refer to the original error just also include the ide settings situation